VirtualReality/Widgets/qMRMLVirtualRealityTransformWidget.cxx

```diff
@@ -0,0 +1,364 @@
+    }
+    else
+    {
+      return vr::TrackingResult_Uninitialized;
+    }
+  }
+}
+//-----------------------------------------------------------------------------
+/// \ingroup Slicer_QtModules_Markups
+class qMRMLVirtualRealityTransformWidgetPrivate
+  : public Ui_qMRMLVirtualRealityTransformWidget
+{
+  Q_DECLARE_PUBLIC(qMRMLVirtualRealityTransformWidget);
+protected:
+  qMRMLVirtualRealityTransformWidget* const q_ptr;
+public:
+  qMRMLVirtualRealityTransformWidgetPrivate(qMRMLVirtualRealityTransformWidget& object);
+  void init();
+
+  vtkWeakPointer<vtkMRMLVirtualRealityViewNode> VRViewNode;
+  vtkWeakPointer<vtkMRMLLinearTransformNode>    TransformNode;
+  vr::ETrackedDeviceClass                       TransformType;
+  vr::ETrackingResult                           PreviousStatus;
+};
+
+//-----------------------------------------------------------------------------
+// qMRMLVirtualRealityTransformWidgetPrivate methods
+
+//-----------------------------------------------------------------------------
+qMRMLVirtualRealityTransformWidgetPrivate::qMRMLVirtualRealityTransformWidgetPrivate(qMRMLVirtualRealityTransformWidget& object)
+  : q_ptr(&object)
+  , VRViewNode(nullptr)
+  , TransformNode(nullptr)
+  , TransformType(vr::TrackedDeviceClass_Invalid)
+  , PreviousStatus(vr::TrackingResult_Uninitialized)
+{
+
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityTransformWidgetPrivate::init()
+{
+  Q_Q(qMRMLVirtualRealityTransformWidget);
+  this->setupUi(q);
+
+  QObject::connect(this->pushButton_Transform, &QPushButton::clicked, q, &qMRMLVirtualRealityTransformWidget::onButtonClicked);
+
+  q->updateWidgetFromMRML();
+}
+
+//-----------------------------------------------------------------------------
+// qMRMLVirtualRealityTransformWidget methods
+
+//-----------------------------------------------------------------------------
+qMRMLVirtualRealityTransformWidget::qMRMLVirtualRealityTransformWidget(QWidget* newParent)
+  : Superclass(newParent)
+  , d_ptr(new qMRMLVirtualRealityTransformWidgetPrivate(*this))
+{
+  Q_D(qMRMLVirtualRealityTransformWidget);
+
+  Q_INIT_RESOURCE(qMRMLVirtualRealityTransformWidget);
+
+  d->init();
+}
+
+//-----------------------------------------------------------------------------
+qMRMLVirtualRealityTransformWidget::~qMRMLVirtualRealityTransformWidget()
+{
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityTransformWidget::setMRMLLinearTransformNode(vtkMRMLNode* node)
+{
+  setMRMLLinearTransformNode(vtkMRMLLinearTransformNode::SafeDownCast(node));
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityTransformWidget::setMRMLLinearTransformNode(vtkMRMLLinearTransformNode* node)
+{
+  Q_D(qMRMLVirtualRealityTransformWidget);
+
+  d->TransformNode = node;
+  if (node == nullptr)
+  {
+    return;
+  }
+
+  findViewNode();
```

Comment:
If you have a call like this that has no error check no return value, then it makes the code harder to read. I'd probably set the member variable from the return value of this function and at the same time check the return value and log error if not found.